src/thermo/PengRobinson.cpp

```diff
@@ -348,119 +348,82 @@ bool PengRobinson::addSpecies(shared_ptr<Species> spec)
-    // open xml file critProperties.xml
-    XML_Node* doc = get_XML_File("critProperties.xml");
-    size_t nDatabase = doc->nChildren();
-
-    // Loop through all species in the database and attempt to match supplied
-    // species to each. If present, calculate pureFluidParameters a_k and b_k
-    // based on crit properties T_c and P_c:
-    for (size_t isp = 0; isp < nDatabase; isp++) {
-        XML_Node& acNodeDoc = doc->child(isp);
-        std::string iNameLower = toLowerCopy(iName);
-        std::string dbName = toLowerCopy(acNodeDoc.attrib("name"));
-
-        // Attempt to match provided species iName to current database species
-        //  dbName:
-        if (iNameLower == dbName) {
-            // Read from database and calculate a and b coefficients
-            double vParams;
-            double T_crit = 0.0, P_crit = 0.0, w_ac = 0.0;
-
-            if (acNodeDoc.hasChild("Tc")) {
-                vParams = 0.0;
-                XML_Node& xmlChildCoeff = acNodeDoc.child("Tc");
-                if (xmlChildCoeff.hasAttrib("value")) {
-                    std::string critTemp = xmlChildCoeff.attrib("value");
-                    vParams = strSItoDbl(critTemp);
-                }
-                if (vParams <= 0.0) { //Assuming that Pc and Tc are non zero.
-                    throw CanteraError("PengRobinson::getCoeff",
-                        "Critical Temperature must be positive");
-                }
-                T_crit = vParams;
-            }
-            if (acNodeDoc.hasChild("Pc")) {
-                vParams = 0.0;
-                XML_Node& xmlChildCoeff = acNodeDoc.child("Pc");
-                if (xmlChildCoeff.hasAttrib("value")) {
-                    std::string critPressure = xmlChildCoeff.attrib("value");
-                    vParams = strSItoDbl(critPressure);
-                }
-                if (vParams <= 0.0) { //Assuming that Pc and Tc are non zero.
-                    throw CanteraError("PengRobinson::getCoeff",
-                        "Critical Pressure must be positive");
-                }
-                P_crit = vParams;
-            }
-            if (acNodeDoc.hasChild("omega")) {
-                vParams = 0.0;
-                XML_Node& xmlChildCoeff = acNodeDoc.child("omega");
-                if (xmlChildCoeff.hasChild("value")) {
-                    std::string acentric_factor = xmlChildCoeff.attrib("value");
-                    vParams = strSItoDbl(acentric_factor);
-                }
-                w_ac = vParams;
-            }
-
-            spCoeff[0] = omega_a * (GasConstant * GasConstant) * (T_crit * T_crit) / P_crit; //coeff a
-            spCoeff[1] = omega_b * GasConstant * T_crit / P_crit; // coeff b
-            spCoeff[2] = w_ac; // acentric factor
-            break;
-        }
-    }
-    // If the species is not present in the database, throw an error
-    if(isnan(spCoeff[0]))
-    {
+    AnyMap data = AnyMap::fromYamlFile("critical-properties.yaml");
```

Comment:
While this isn’t something that was introduced here, this means that the entire file is being parsed any time coefficients are needed. Is there a more efficient way to accomplish this (e.g. by buffering the file content the first time the file is read?)

Reply:
This isn't quite as bad as it looks -- `AnyMap::fromYamlFile` caches the `AnyMap` generated for any given input file, and only re-reads it from disk if it's been modified. Changing this is a bit of scope creep for this PR, but I think solving it (for both P-R and R-K) won't be too complicated.

Reply:
If the resulting `AnyMap` is indeed cached (i.e. `yaml-cpp` does not have to parse again), then this observation could be laid to rest by including a comment (i.e. no other action needed).